src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -369,21 +376,28 @@ public static void updateSolidTransactions(final Tangle tangle, final Set<Hash>
         while(hashIterator.hasNext()) {
             transactionViewModel = TransactionViewModel.fromHash(tangle, hashIterator.next());
             transactionViewModel.updateHeights(tangle);
-            transactionViewModel.updateSolid(true);
+            transactionViewModel.updateSolid(SubtangleStatus.SOLID);
             transactionViewModel.update(tangle, "solid|height");
         }
     }
 
-    public boolean updateSolid(boolean solid) throws Exception {
-        if(solid != transaction.solid) {
-            transaction.solid = solid;
+    public boolean updateSolid(SubtangleStatus status) throws Exception {
```

Comment:
maybe worth changing the name now that it's not only solid - true / false:
`updateStatus` / `updateSubtangleStatus` / `updateSolidStatus` ?